cpp/example_code/iam/tests/iam_gtests.cpp

```diff
@@ -19,13 +19,62 @@
 #include <aws/iam/model/CreatePolicyRequest.h>
 #include <aws/iam/model/CreateAccountAliasRequest.h>
 #include <aws/core/utils/UUID.h>
+#include <fstream>
 
 Aws::SDKOptions AwsDocTest::IAM_GTests::s_options;
 std::unique_ptr<Aws::Client::ClientConfiguration> AwsDocTest::IAM_GTests::s_clientConfig;
 Aws::String AwsDocTest::IAM_GTests::s_accessKey;
 Aws::String AwsDocTest::IAM_GTests::s_role;
 Aws::String AwsDocTest::IAM_GTests::s_userName;
 Aws::String AwsDocTest::IAM_GTests::s_policyArn;
+static const char ALLOCATION_TAG[] = "IAM_GTEST";
+
+/*
+ * Subclass MockHTTPCLient to respond to credential requests.
+ * Otherwise, the stored responses are returned for credential requests
+ * and not the service API calls.
+ */
+class CustomMockHTTPClient : public MockHttpClient {
+public:
+    explicit CustomMockHTTPClient(
+            const std::shared_ptr<Aws::Http::HttpRequest> &requestTmp) {
+        std::shared_ptr<Aws::Http::Standard::StandardHttpResponse> goodResponse = Aws::MakeShared<Aws::Http::Standard::StandardHttpResponse>(
+                ALLOCATION_TAG, requestTmp);
+        goodResponse->AddHeader("Content-Type", "text/json");
+        goodResponse->SetResponseCode(Aws::Http::HttpResponseCode::OK);
+        Aws::Utils::DateTime expiration =
+                Aws::Utils::DateTime::Now() + std::chrono::milliseconds(60000);
+
+        goodResponse->GetResponseBody() << "{"
+                                        << R"("RoleArn":"arn:aws:iam::123456789012:role/MockRole",)"
```

Comment:
Reminder that https://alpha.www.docs.aws.a2z.com/awsstyleguide/latest/styleguide/safenames.html#resourceID exists.

Reply:
Actually, it is fictionalized, i.e. sequential numbers. Just not hexadecimal. I start over at 9. I have been using this consistently for an ID for a while.

Reply:
This is a valid fictionalized account ID.
https://alpha.www.docs.aws.a2z.com/awsstyleguide/latest/styleguide/safenames.html#resourceID:~:text=%26ExampleAWSAccountNo2H%3B)-,123456789012,-(%26ExampleAWSAccountNo3%3B